mat/list_test.go

```diff
@@ -625,6 +661,8 @@ var testMatrices = []Matrix{
 	&DiagDense{},
 	&DiagDense{mat: blas64.Vector{Inc: 1}},
 	&DiagDense{mat: blas64.Vector{Inc: 10}},
+	NewTriBandDense(6, 2, Upper, nil),
```

Comment:
Zero values here too?

Reply:
The note above is
`// The TriDense types have actual sizes because the return from Triangular is
// only valid when n == 0.`

We intentionally corrupt the return type when n is zero. We can fix this by not doing this corruption, for example changing the call to `Triangle` to be `return t.mat.N, t.triKind()`. This is simpler, and either way the user is warned not to rely on the value. If we want to keep the corruption, then we have to make the Triangular non-zero valued.

Reply:
OK, we might want to revisit this then, but not here.